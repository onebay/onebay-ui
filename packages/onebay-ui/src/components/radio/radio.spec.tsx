import { mount } from '@vue/test-utils'
import Radio from './index'

const options = [
  { label: 'Radio one', value: 'option1' },
  { label: 'Radio two', value: 'option2', desc: 'Radio description two' },
  { label: 'Radio three', value: 'option3', desc: 'Radio description three', disabled: true },
]

describe('Radio Snap', () => {
  it('render Radio', () => {
    const wrapper = mount({
      render() {
        return <Radio options={options} />
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Radio default options', () => {
    const wrapper = mount({
      render() {
        return <Radio />
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Radio Event', () => {
  it('Radio onClick', () => {
    const onItemClick = jest.fn()
    const wrapper = mount({
      render() {
        return <Radio value="option2" options={options} onClick={onItemClick} />
      },
    })
    wrapper.find('.ob-radio .ob-radio__option').trigger('click')
    expect(onItemClick).toBeCalled()
    expect(onItemClick.mock.calls[0][0]).toBe('option1')
  })

  it('Radio onClick disabled, onClick not to be called', () => {
    const onItemClick = jest.fn()
    const wrapper = mount({
      render() {
        return <Radio value="option2" options={options} onClick={onItemClick} />
      },
    })
    wrapper.find('.ob-radio .ob-radio__option:nth-child(3)').trigger('click')
    expect(onItemClick).not.toBeCalled()
  })
})
