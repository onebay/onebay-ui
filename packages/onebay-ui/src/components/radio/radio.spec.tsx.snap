// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Snap render Radio 1`] = `
<div
  class="ob-radio"
>
  
  <div
    class="ob-radio__option"
  >
    <div
      class="ob-radio__option-wrap"
    >
      <div
        class="ob-radio__option-container"
      >
        <div
          class="ob-radio__title"
        >
          Radio one
        </div>
        <div
          class="ob-radio__icon"
        >
          <div
            class="ob-icon ob-icon-check text"
          />
        </div>
      </div>
      <!---->
    </div>
  </div>
  <div
    class="ob-radio__option"
  >
    <div
      class="ob-radio__option-wrap"
    >
      <div
        class="ob-radio__option-container"
      >
        <div
          class="ob-radio__title"
        >
          Radio two
        </div>
        <div
          class="ob-radio__icon"
        >
          <div
            class="ob-icon ob-icon-check text"
          />
        </div>
      </div>
      <div
        class="ob-radio__desc"
      >
        Radio description two
      </div>
    </div>
  </div>
  <div
    class="ob-radio__option ob-radio__option--disabled"
  >
    <div
      class="ob-radio__option-wrap"
    >
      <div
        class="ob-radio__option-container"
      >
        <div
          class="ob-radio__title"
        >
          Radio three
        </div>
        <div
          class="ob-radio__icon"
        >
          <div
            class="ob-icon ob-icon-check text"
          />
        </div>
      </div>
      <div
        class="ob-radio__desc"
      >
        Radio description three
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Radio Snap render Radio default options 1`] = `
<div
  class="ob-radio"
>
  
  
</div>
`;
